runner/app/live/trickle/media.py

```diff
@@ -149,36 +93,47 @@ async def callback(pipe_file, pipe_name):
                     await segment.write(data)
                 transport.close()
 
-        def sync_callback(pipe_fd, pipe_name):
-            # asyncio.connect_read_pipe expects explicit fd close
-            # so we have to manually read, detect eof, then close
-            r, w = os.pipe()
-            rf = os.fdopen(r, 'rb', buffering=0)
-            future = asyncio.run_coroutine_threadsafe(callback(rf, pipe_name), loop)
-            try:
-                while True:
-                    data = pipe_fd.read(32 * 1024)
-                    if not data:
-                        break
-                    os.write(w, data)
-                os.close(w) # streamreader is very sensitive about this
-                future.result()  # This blocks in the thread until callback completes
-                rf.close() # also closes the read end of the pipe
-            # Ensure any exceptions in the coroutine are caught
-            except Exception as e:
-                logging.error(f"Error in sync_callback: {e}")
-
-        ffmpeg_read_fd, ffmpeg_write_fd = os.pipe()
-        segment_thread = threading.Thread(target=segmenter.segment_reading_process, args=(ffmpeg_read_fd, sync_callback))
-        ffmpeg_feeder = threading.Thread(target=feed_ffmpeg, args=(ffmpeg_write_fd, image_generator))
-        segment_thread.start()
-        ffmpeg_feeder.start()
-        logging.debug("run_publish: ffmpeg feeder and segmenter threads started")
-
+        def sync_callback(pipe_file, pipe_name):
+            def do_schedule():
+                schedule_callback(callback(pipe_file, pipe_name), pipe_name)
+            loop.call_soon_threadsafe(do_schedule)
+
+        # hold tasks since `loop.create_task` is a weak reference that gets GC'd
+        # TODO use asyncio.TaskGroup once all pipelines are on Python 3.11+
```

Comment:
Can we not switch to python 3.11 right now? Is there a limitation? The logic here looks alright but it feels like using TaskGroup would simplify it a bit to avoid manual tracking of tasks. We can merge liek this for now and re-test but perhaps should consider revisiting before launch if we see issues? wdyt? 

cc @victorges as well for comments on this since I'm still relearning Pythonic concurrency.

Reply:
I believe there are dependency limitations in some pipelines that preclude us from using newer Python although I am not sure what the minimum version would be - @victorges might know better. Didn't want to introduce the risk of a Python upgrade and pipeline validation alongside this PR which is already invasive enough as-is